rust/src/traits/samplers/uniform/mod.rs

```diff
@@ -156,7 +154,7 @@ macro_rules! impl_sample_uniform_unsigned_int {
                 loop {
                     // algorithm is only valid when sample_uniform_int is non-negative
                     let v = Self::sample_uniform_int()?;
-                    if v <= Self::MAX - Self::MAX % upper {
+                    if v < Self::MAX - Self::MAX % upper {
```

Comment:
Was the `<=` incorrect before? Not sure how this relates to the comment above.

Reply:
It was incorrect before. I've written a draft of a proof and opened a separate PR for it.